adagrams/game.py

```diff
@@ -1,11 +1,79 @@
+import random
+from collections import Counter
+LETTER_POOL20 = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
 def draw_letters():
-    pass
+    values = list(LETTER_POOL20.values())
+    keys = list(LETTER_POOL20.keys())
+    players_hand =random.sample(keys, counts = values, k =10)
+    return players_hand
 
 def uses_available_letters(word, letter_bank):
-    pass
-
+    word = word.upper()
+    available_bank = Counter(letter_bank)
+    for letter in word:
+        if letter in letter_bank and available_bank[letter]:
+            available_bank[letter] -= 1
+            available = True
+        else:
+            return False
+    return available
+            
 def score_word(word):
-    pass
-
+    word = word.upper()
+    point_system = {
+        ("A", "E", "I", "O", "U", "L", "N", "R", "S", "T") : 1,
+        ("D","G"): 2,
+        ("B", "C", "M", "P"):3,
+        ("F", "H", "V", "W", "Y"):4,
+        ("K"): 5,
+        ("J","X"):8,
+        ("Q","Z"):10
+    }
+    score = 0
+    for letter in word:
+        for key in point_system:
+            if letter in key:
```

Comment:
You've used tuples as keys and have the points as values. You've already got one loop on line 60 and then when you have `if letter in key` you introduce another operation that takes linear time because the `in` operator for a tuple needs to visit each element to see if key is actually in it. 

While the data structure would look longer, if you converted point_system to a dict with each key being a single letter and value is points, then you can leverage the fact that dictionaries have constant look up time and reduce time complexity.